v2/tools/generator/internal/interfaces/kubernetes_resource_interface.go

```diff
@@ -71,6 +72,9 @@ func AddKubernetesResourceInterfaceImpls(
 	getResourceScopeFunction := functions.NewObjectFunction("GetResourceScope", idFactory, newGetResourceScopeFunction(r))
 	getResourceScopeFunction.AddPackageReference(astmodel.GenRuntimeReference)
 
+	getSupportedOperationsFunction := functions.NewObjectFunction("GetSupportedOperations", idFactory, newGetSupportedOperationsFunction(r))
```

Comment:
Fixed. Reason I didn't do this originally is it's not what the other functions in this method were doing and I just copied them.

I've updated the existing functions (that can easily move to use `NewResourceFunction`) to do so here as well so that the slightly easier pattern is copied in the future.